### DISCUSSION TITLE
Question is it possible to read GML 3.2

### DISCUSSION THREAD

**Discussion by swidz** [NetTopologySuite/NetTopologySuite#539] [Q&A]
I would like to ask if current version is cappable of reading gml 3.2.

I have an xml document describing administrative areas, which uses Gml 3.2 to describe geometry.

From the docs I see that reading and writing is possible for gml 2 http://nettopologysuite.github.io/NetTopologySuite/api/NetTopologySuite.IO.GML2.GMLReader.html and only writing for gml http://nettopologysuite.github.io/NetTopologySuite/api/NetTopologySuite.IO.GML3.html but no reader class for ver 3

However, based on the issue history, I see that there was some work done on reading gml version 3 https://github.com/NetTopologySuite/NetTopologySuite/pull/472 

Is it coded in version 2 of gmlreader?

**Comment by airbreather** [comment]
`NetTopologySuite.IO.GML2.GMLReader` will read GML2 if that's what you give it, or GML3 if that's what you give it.  The namespace is slightly unfortunate.

When writing, you have to pick one or the other, which is why `NetTopologySuite.IO.GML2.GMLWriter` and `NetTopologySuite.IO.GML3.GML3Writer` are separate.
